src/Raven.Server/Rachis/RachisConsensus.cs

```diff
@@ -1116,11 +1117,18 @@ public void AcceptNewConnection(Stream stream, Action disconnect, EndPoint remot
             }
         }
 
-        private static void ValidateCompatibility(RachisHello initialMessage)
+        private void ValidateCompatibility(RachisHello initialMessage, ClusterTopology clusterTopology)
         {
             var version = initialMessage.ServerBuildVersion;
-            if (version == 0 || ServerVersion.IsNightlyOrDev(version))  
+            if (ServerVersion.IsNightlyOrDev(version))
                 return;
+
+            if (clusterTopology.Members.ContainsKey(Tag) == false)
+            {
+                if (version >= 40_000 && version < 42_000)
```

Comment:
Since we cannot distinguish between 4.1 and 4.2.0 we will allow then, but all new 4.1.x builds will be blocked.